delfin/task_manager/tasks/resources.py

```diff
@@ -239,3 +240,97 @@ def sync(self):
     def remove(self):
         LOG.info('Remove volumes for storage id:{0}'.format(self.storage_id))
         db.volume_delete_by_storage(self.context, self.storage_id)
+
+
+class PerformanceCollectionTask(object):
+
+    def __init__(self, context, storage_id, interval, is_history):
+        self.context = context
+        self.storage_id = storage_id
+        self.interval = interval
+        self.is_history = is_history
+        self.driver_api = driverapi.API()
+        self.exporter_api = exporterapi.API()
+
+
+class ArrayPerformanceCollection(PerformanceCollectionTask):
+    def __init__(self, context, storage_id, interval, is_history):
+        super(ArrayPerformanceCollection, self).__init__(context, storage_id,
+                                                         interval, is_history)
+
+    def collect(self):
+        """
+        :return:
+        """
+        LOG.info('Collecting array performance metrics for storage id:{0}'
+                 .format(self.storage_id))
+        try:
+            # collect the performance metrics from driver and push to
+            # prometheus exporter api
+            storage_metrics = self.driver_api.collect_array_metrics(
+                self.context, self.storage_id, self.interval, self.is_history)
+
+            self.exporter_api.push_storage_metrics(storage_metrics)
```

Comment:
Just  Export to base exporter which handles dispatch for all exporters, just like how we do laert exporting